### PR TITLE
Opening member-details panel should not open/close grid view

### LINKED ISSUE #457: Opening member details panel opens/closes grid view

### PR DESCRIPTION
**Pull request by MidhunSureshR**
Fixes #457 
Signed-off-by: RMidhunSuresh <rmidhunsuresh@gmail.com>

### CODE REVIEW

File: src/domain/navigation/index.js [hunk 1]
```diff
@@ -132,11 +132,14 @@ export function parseUrlPath(urlPath, currentNavPath, defaultSessionId) {
                 segments.push(roomsSegmentWithRoom(rooms, roomId, currentNavPath));
             }
             segments.push(new Segment("room", roomId));
-            // Add right-panel segments from previous path
-            const previousSegments = currentNavPath.segments;
-            const i = previousSegments.findIndex(s => s.type === "right-panel");
-            if (i !== -1) {
-                segments.push(...previousSegments.slice(i));
+            if (!urlPath.includes("/member/")) {
```
Comment (bwindels) [inline src/domain/navigation/index.js]: Hah, I completely misunderstood what you are doing here for some reason likely more related to me than to your code, to then look into the code wondering if we couldn't reuse the `open-room` action and come to exactly the same conclusion as you to then come back here and suggest that and go ... aaaaah, that's what's happening here :smile: 

Having said that, I wonder if we can make the check a bit more generic here by just checking if  there is only one element (the room id) after the `open-room` part in `parts`. Wdyt?

File: src/domain/session/rightpanel/MemberTileViewModel.js [hunk 2]
```diff
@@ -47,7 +48,7 @@ export class MemberTileViewModel extends ViewModel {
     }
 
     get detailsUrl() {
-        return `${this.urlCreator.urlUntilSegment("room")}/member/${this._member.userId}`;
+        return `${this.urlCreator.openRoomActionUrl(this._roomId)}/member/${this._member.userId}`;
```
Comment (bwindels) [inline src/domain/session/rightpanel/MemberTileViewModel.js]: you could also get the current room id from `this.navigation.path.get("room").value` so you don't need to inject it ...

**Review by bwindels — COMMENTED** [review]
Took me a while to come to the same conclusion as you, be great approach :smile: perhaps we can make the check in the parseUrl function a bit more generic though.

File: src/domain/navigation/index.js (hunk 1 above)
Comment (MidhunSureshR) [reply src/domain/navigation/index.js]: Sounds good to me :+1:

File: src/domain/session/rightpanel/MemberTileViewModel.js (hunk 2 above)
Comment (MidhunSureshR) [reply src/domain/session/rightpanel/MemberTileViewModel.js]: Oops!🤦‍♂️

**Comment by MidhunSureshR** [comment]
~~I should also write a test for the new logic in `index.js`...~~ Done

File: src/domain/navigation/index.js [hunk 3]
```diff
@@ -132,8 +132,10 @@ export function parseUrlPath(urlPath, currentNavPath, defaultSessionId) {
                 segments.push(roomsSegmentWithRoom(rooms, roomId, currentNavPath));
             }
             segments.push(new Segment("room", roomId));
-            if (!urlPath.includes("/member/")) {
-                // Add right-panel segments from previous path if /member is not in url
+            const partIndex = parts.findIndex(part => part === "open-room");
+            const partsAfterRoom = parts.slice(partIndex + 1);
+            if (partsAfterRoom.length === 1) {
```
Comment (bwindels) [inline src/domain/navigation/index.js]: Thanks for the changes! A bit of a nit, but I don't think we need to create an array here with slice, as we won't be using it. Can also do just:
```js
            const openRoomPartIndex = parts.findIndex(part => part === "open-room");
            // -2 for the room id after open-room and the open-room part itself
            const hasOnlyRoomIdAfterPart = openRoomPartIndex >= parts.length - 2;
            if (hasOnlyRoomIdAfterPart) {
```

Comment (MidhunSureshR) [reply src/domain/navigation/index.js]: Done :heavy_check_mark:

**Review by bwindels — APPROVED** [review]
Thanks!

**Pull request merged**